src/Symfony/Component/EventDispatcher/EventSubscriberInterface.php

```diff
@@ -43,7 +43,7 @@ interface EventSubscriberInterface
      * The code must not depend on runtime state as it will only be called at compile time.
      * All logic depending on runtime state must be put into the individual methods handling the events.
      *
-     * @return array The event names to listen to
+     * @return array<string, mixed> The event names to listen to
```

Comment:
Can we use something more explicit than mixed? Personally I try to use union's as much as possible.
Like `array<string, string|array{0: string}|array{0: string, 1: int}}`

Reply:
that's actually not the right signature. the right one is this:

```
array<string, string|array{0: string, 1?: int}|list<array{0: string, 1?: int}>>
```

Reply:
OK, you edited in the meantime. Your new proposal is too verbose due to not using the optional key for the priority

Reply:
This notation is not understood by all tools yet. I think the current type is precise enough.

Reply:
TIL the list pseudo-type 😄 and optional keys.

Reply:
https://psalm.dev/r/0e971e1c02 but for some reason it doesn't like it :D

Reply:
You can't use `@template` for this (as that's a generic and implies the return type), you need to define a custom type instead but Pslam doesn't support this it seems.

I only use PHPStan atm because Psalm doesn't support PHP 8 yet 😞